### DISCUSSION TITLE
Dimenet_QM9 pretrained error

### DISCUSSION THREAD

**Discussion by sudaksh14** [pyg-team/pytorch_geometric#5894] [Q&A]
Hi PyG Community,
I was trying to load the pretrained dimenet model from PyG but faced with the below error:

![image](https://user-images.githubusercontent.com/102596407/199842988-470c6eff-9369-4df9-8e31-168141b75437.png)

Could you pls suggest what could be missing as I have already installed the latest version of tensorflow in the environment.

**Comment by rusty1s** [comment]
Which version of `tensorflow` are you using? `tf.train.load_checkpoint` should exist, see https://www.tensorflow.org/api_docs/python/tf/train/load_checkpoint.

**Reply by sudaksh14** [reply]
Hi Matthias,
I installed the latest version 2.10.0 but as per documentation it also contains this function.
Thnx

**Reply by rusty1s** [reply]
What happens if you run
```
import tensorflow as tf
print(tf.train.load_checkpoint)
```

**Reply by sudaksh14** [reply]
I am not sure whats the issue with tensorflow it was working fine initially, now its giving this error:
![image](https://user-images.githubusercontent.com/102596407/199958300-b55d9ee4-b695-4184-a520-822d233c3156.png)

Even with pip list it shows wrapt is installed.
![image](https://user-images.githubusercontent.com/102596407/199958514-5a6e4c1b-d98a-46ae-a561-338a45903dd8.png)

**Reply by sudaksh14** [reply]
I have created a virtual environment for my project in which I am working mostly with pytorch but on root environment it shows the foll result for above command:

![image](https://user-images.githubusercontent.com/102596407/199959088-099cbfed-af52-4a87-9cdd-10d9b5402b2e.png)

**Reply by rusty1s** [reply]
This is super weird. If the above command works, I don't see any reason why it shouldn't work in PyG. Does the error still exist by now?

**Reply by sudaksh14** [reply]
Yes it still exists. I think I should try to remove tf from virtual env and install again a lower version of tf.

**Reply by sudaksh14** [reply]
Hi Matthias, I downgraded the tf to 2.9.1 and then this was the error again:

![image](https://user-images.githubusercontent.com/102596407/201354673-456e1108-00a6-4c7b-880b-74892d444ea5.png)

Does this has anything to do with license? I am not sure according to pip list following packages are already installed and wrapt is present as well as tf.

absl-py                      1.3.0
asttokens                    2.0.5
astunparse                   1.6.3
backcall                     0.2.0
Bottleneck                   1.3.5
brotlipy                     0.7.0
cachetools                   5.2.0
certifi                      2022.9.14
cffi                         1.15.1
charset-normalizer           2.0.4
colorama                     0.4.5
cryptography                 37.0.1
cycler                       0.11.0
debugpy                      1.5.1
decorator                    5.1.1
entrypoints                  0.4
executing                    0.8.3
flatbuffers                  1.12
fonttools                    4.25.0
gast                         0.4.0
google-auth                  2.14.0
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
grpcio                       1.50.0
h5py                         3.7.0
idna                         3.3
importlib-metadata           5.0.0
ipykernel                    6.15.2
ipython                      8.4.0
jedi                         0.18.1
Jinja2                       3.0.3
joblib                       1.1.0
jupyter_client               7.3.5
jupyter-core                 4.10.0
keras                        2.9.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.2
libclang                     14.0.6
Markdown                     3.4.1
MarkupSafe                   2.1.1
matplotlib                   3.5.2
matplotlib-inline            0.1.6
mkl-fft                      1.3.1
mkl-random                   1.2.2
mkl-service                  2.4.0
mpmath                       1.2.1
munkres                      1.1.4
nest-asyncio                 1.5.5
networkx                     2.8.4
numexpr                      2.8.3
numpy                        1.22.4
oauthlib                     3.2.2
opt-einsum                   3.3.0
packaging                    21.3
pandas                       1.4.4
parso                        0.8.3
pickleshare                  0.7.5
Pillow                       9.2.0
pip                          22.1.2
ply                          3.11
prompt-toolkit               3.0.20
protobuf                     3.19.6
psutil                       5.9.0
pure-eval                    0.2.2
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycparser                    2.21
Pygments                     2.11.2
pyOpenSSL                    22.0.0
pyparsing                    3.0.9
PyQt5                        5.15.7
PyQt5-sip                    12.11.0
PySocks                      1.7.1
python-dateutil              2.8.2
python-louvain               0.15
pytz                         2022.1
pywin32                      302
PyYAML                       6.0
pyzmq                        23.2.0
requests                     2.28.1
requests-oauthlib            1.3.1
rsa                          4.9
scikit-learn                 1.1.1
scipy                        1.7.1
setuptools                   63.4.1
sip                          6.6.2
six                          1.16.0
stack-data                   0.2.0
sympy                        1.10.1
tensorboard                  2.9.1
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.9.0
tensorflow-estimator         2.9.0
tensorflow-io-gcs-filesystem 0.27.0
termcolor                    2.1.0
tf-estimator-nightly         2.8.0.dev2021122109
threadpoolctl                2.2.0
toml                         0.10.2
torch                        1.11.0
torch-cluster                1.6.0
torch-geometric              2.1.0
torch-scatter                2.0.9
torch-sparse                 0.6.14
torch-spline-conv            1.2.1
tornado                      6.2
tqdm                         4.64.0
traitlets                    5.1.1
typing_extensions            4.3.0
urllib3                      1.26.11
wcwidth                      0.2.5
Werkzeug                     2.2.2
wheel                        0.37.1
win-inet-pton                1.1.0
wincertstore                 0.2
wrapt                        1.12.1
yacs                         0.1.6
zipp                         3.10.0

**Reply by EdisonLeeeee** [reply]
How about reinstalling wrapt?
```bash
pip uninstall wrapt -y
pip install wrapt
```

**Reply by sudaksh14** [reply]
Hi Edison,
It worked thnx

**Reply by sudaksh14** [reply]
Hi Matthias and Edison,
I came up with another issue with using the pre-trained dimenet model which is as below:

![image](https://user-images.githubusercontent.com/102596407/205172372-da3c0fda-2bb2-4508-8685-9e1cb040db9e.png)

Here the error seems that the original dimenet was written with tf but pytorch geometric is based upon pytorch. Could you suggest any solution.
Thanks for your help and time!

**Reply by rusty1s** [reply]
Is it possible for you to print `train_idx` in `nn/models/dimenet.py` line 569? What's the output here?

**Reply by sudaksh14** [reply]
Hi Matthias,
Below is the snapshot of output:

![image](https://user-images.githubusercontent.com/102596407/205318909-407cddb5-b8bb-4fcd-aa3f-0c297ab6d690.png)

Its strange that the type is torch.tensor, still the error above says that torch.tensor type is acceptable.

**Reply by EdisonLeeeee** [reply]
`torch.int32` is not supported as an index, you need to convert it to `torch.long` instead, e.g., `x.long()`.

**Reply by sudaksh14** [reply]
Thnx Edison,
It works now but I downloaded the QM9 data from the pytorch geometric library, so I think it could be updated to implement the long tensor as the data type.
![image](https://user-images.githubusercontent.com/102596407/205444954-8c1651ac-733e-40a6-9562-1b6e8def5d3d.png)
